### PR TITLE
Data helpers to abstract curated/backfill differences

### PR DESCRIPTION
**Pull request by andre1050**
## What does this change?
Adds a `dataHelper` file which exports a few helper functions that drill through the story object looking for deeply nested properties, or properties which we only want to use conditionally. This abstracts properties that are specific to `curated` or `backfill` stories so that each card consuming them can be dropped in any collection regardless of whether it's curated of backfill.

This cleans up a significant amount of code by centralising in one place the logic to extract `kicker`, `byline`, `imageSrc`, `pillar` and `cardUrl` from any given story object.

## Why?
To ensure neither the collections not the cards themselves are coupled to the type of story in use (curated or backfill). An added benefit is a cleaner interface for the components by not having to check if the data actually exists when it's passed downstream.

### CODE REVIEW

File: src/collections/GenericCollection.tsx [hunk 1]
```diff
@@ -9,7 +9,8 @@ import { Multiline } from "../components/Multiline";
 export const GenericCollection: React.FC<{
     collection: ICollection;
 }> = ({ collection }) => {
-    if (collection.backfill.length < 1) {
+    const content = [].concat(collection.curated).concat(collection.backfill);
```
Comment (nicl) [inline src/collections/GenericCollection.tsx]: Think you can just do:

    [].contact(a, b)

i.e. concat is multi-arity.

File: src/collections/InstagramCollection.tsx [hunk 2]
```diff
@@ -5,15 +5,22 @@ import { Padding } from "../layout/Padding";
...
+    getPillarName,
+    getImageSrc,
+    getCardUrl
+} from "../dataHelpers";
 
 export const InstagramCollection: React.FC<{
     collection: ICollection;
 }> = ({ collection }) => {
-    const content = collection.curated.concat(collection.backfill);
+    const content = [].concat(collection.curated).concat(collection.backfill);
```
Comment (nicl) [inline src/collections/InstagramCollection.tsx]: Ibid.

File: src/dataHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        return null;
+    }
+};
+
+export const getCardUrl = (story: Content): string => {
+    const baseUrl = story.properties.webUrl
+        ? story.properties.webUrl
+        : `https://www.theguardian.com${story.properties.href}`;
+
+    const brazeParam = "?##braze_utm##";
```
Comment (nicl) [inline src/dataHelpers.ts]: Is it worth keeping this in the components themselves? I.e. at the point of use? My thinking is that it is not a core part of the URL/model so there may be cases in the future where we want the URL without the braze param. Instead, we could create a helper function such as `withBrazeParam` to wrap URLs within components?

It's not a blocker - I can see that this saves a lot of typing. Just I can see it being potentially confusing/restrictive later down the line to modify the core model in this way.

File: src/collections/GenericCollection.tsx (hunk 1 above)
Comment (andre1050) [reply src/collections/GenericCollection.tsx]: Yeah of course 🤦‍♂️

File: src/dataHelpers.ts (hunk 3 above)
Comment (andre1050) [reply src/dataHelpers.ts]: Good point, I just didn't find any use case where we wouldn't want the cardUrl to have the parameter as it's also used for click-thru links - but easy to update with a `withBrazeParam` and be more future-proof.

**Pull request merged**